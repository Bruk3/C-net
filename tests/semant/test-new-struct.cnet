struct Dimension {
    float width;
    float height;
};

struct Shape {
    struct Dimension dimension;
    string type;
};

int main(int argc) {
    struct Shape circle = new struct Shape;
    circle.type = "circle";
    circle.dimension.width = 5.;
}