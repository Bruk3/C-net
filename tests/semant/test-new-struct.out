struct Dimension {
	float width;
	float height;
};

struct Shape {
	struct Dimension dimension;
	string type;
};

int main()
{
{
struct Shape circle = (struct Shape : );
(struct Shape : circle=(struct Shape : new Shape));
}
(string : circle.type=(string : cnet_strcpy((string : circle.type), (string : "circle"))));
(float : circle.dimension.width=(float : 5.));
return (int : 0);
}

